app.js

```diff
@@ -249,44 +277,292 @@ const app = Vue.createApp({
+            score_eq1_next_lower_macro = lookuptable[eq1_next_lower_macro]
+            score_eq2_next_lower_macro = lookuptable[eq2_next_lower_macro]
+
+            if (eq3==0 && eq6==0){
+                //multiple path take the one with higher score
+                score_eq3eq6_next_lower_macro_left = lookuptable[eq3eq6_next_lower_macro_left]
+                score_eq3eq6_next_lower_macro_right = lookuptable[eq3eq6_next_lower_macro_right]
+
+                if (score_eq3eq6_next_lower_macro_left>score_eq3eq6_next_lower_macro_right){
+                    score_eq3eq6_next_lower_macro = score_eq3eq6_next_lower_macro_left
+                }
+                else{
+                    score_eq3eq6_next_lower_macro = score_eq3eq6_next_lower_macro_right
+                }
             }
-            else if(lookup[0] == "1") {
-                value_0 = parseFloat(this.cvssLookupData["0" + lookup.slice(1)])
-                value = Math.min(value_0, value + AV_diff[this.m("AV")] + PR_diff[this.m("PR")] + UI_diff[this.m("UI")] - 0.2)
+            else{
+                score_eq3eq6_next_lower_macro = lookuptable[eq3eq6_next_lower_macro]
             }
-            else if(lookup[0] == "2") {
-                value_1 = parseFloat(this.cvssLookupData["1" + lookup.slice(1)])
-                value = Math.min(value_1, value + AV_diff[this.m("AV")] + PR_diff[this.m("PR")] + UI_diff[this.m("UI")])
+
+
+            score_eq4_next_lower_macro = lookuptable[eq4_next_lower_macro]
+            score_eq5_next_lower_macro = lookuptable[eq5_next_lower_macro]
+
+            //get all max vector for the eq
+            eq1_maxes = this.getvalueEqLookup(lookup,0)
+            eq2_maxes = this.getvalueEqLookup(lookup,1)
+            eq3_eq6_maxes = this.getvalueEqLookup(lookup,2)[lookup[5]]
+            eq4_maxes = this.getvalueEqLookup(lookup,3)
+            eq5_maxes = this.getvalueEqLookup(lookup,4)
+
+            //compose them
+            max_vectors = []
+            for (eq1_max of eq1_maxes){
+                for (eq2_max of eq2_maxes){
+                    for (eq3_eq6_max of eq3_eq6_maxes){
+                        for (eq4_max of eq4_maxes){
+                            for (eq5max of eq5_maxes){
+                                    max_vectors.push(eq1_max+eq2_max+eq3_eq6_max+eq4_max+eq5max)
+                            }
+                        }
+                    }
+                }
             }
 
-            // EQ2 Min Score Differential
-            AC_diff={"L": 0.1, "H": 0}
-            AT_diff={"N": 0.1, "P": 0}
+            if (max_vectors==undefined){
+                return "0.0"
+            }
+
+
+            // compute hamming distance
+            for (let i = 0; i < max_vectors.length; i++) {
+                tmp_vector = max_vectors[i]
+                //cannot have a negative distance if less than max
+                hamming_distance_AV = AV_levels[this.m("AV")]-AV_levels[this.extractValueMetric("AV",tmp_vector)]
```

Comment:
This seems very complex and repeated similar code. This does not look like proper implementation. Seems like some kind of loop and better data structure may be more readable. It is hard to judge without proper computational requirements.